src/geocat/viz/contourf/_plot_util.py

```diff
@@ -1,168 +1,387 @@
+            righttitle (:obj:`str`): Title for top right subtitle.
+
+            righttitlefontsize (:obj:`int`): Font size for top right subtitle. Default 18.
+
+            ticklabelfontsize (:obj:`int`): Font size of x and y axis ticks. Default 16.
+
+            xlabel (:obj:`str`): Label for the x axis.
+
+            ylabel (:obj:`str`): Label for the y axis.
+        """
+
+        # Pull out titles and labels and their font arguments
         self.maintitle = kwargs.get('maintitle')
+        self.maintitlefontsize = kwargs.get('maintitlefontsize')
         self.lefttitle = kwargs.get('lefttitle')
+        self.lefttitlefontsize = kwargs.get('lefttitlefontsize')
         self.righttitle = kwargs.get('righttitle')
+        self.righttitlefontsize = kwargs.get('righttitlefontsize')
+        self.xlabel = kwargs.get('xlabel')
+        self.ylabel = kwargs.get('ylabel')
+        self.labelfontsize = kwargs.get('labelfontsize')
 
-        # Pull out axes info
-        self.xlim = kwargs.get('xlim')
-        self.ylim = kwargs.get('ylim')
+        # Pull out colorbar arguments
+        self.cbar = None
+        self.individual_cb = kwargs.get("individual_cb")
+        self.add_colorbar = kwargs.get('add_colorbar')
+        self.mappable = kwargs.get('mappable')
+        self.cborientation = kwargs.get('cborientation')
+        self.cbshrink = kwargs.get('cbshrink')
+        self.cbpad = kwargs.get('cbpad')
+        self.cbdrawedges = kwargs.get('cbdrawedges')
+        self.cbticks = kwargs.get('cbticks')
+        self.cbextend = kwargs.get('cbextend')
+        self.cbticklabels = kwargs.get("cbticklabels")
+        self.cbticklabelsize = kwargs.get("cbticklabelsize")
 
-        # Make sure x and y limits specified (for now)
-        #TODO: make x and y limits self-determinable somehow
-        if self.xlim is None or self.ylim is None:
-            raise AttributeError(
-                "For now, xlim and ylim must be specified as kwargs")
+        # Set up figure and axes with specified format
+        _fig_ax._init_(self, *args, **kwargs)
 
-        # Pull out tick arguments if specified
-        self.xticks = kwargs.get('xticks')
-        self.yticks = kwargs.get('yticks')
+        # Add geographical features to figure
+        add_geo_features._init_(self, *args, **kwargs)
 
-        # Pull out axes label arguments
-        self.xlabel = kwargs.get('xlabel')
-        self.ylabel = kwargs.get('ylabel')
+        # Add titles to figure
+        self.add_titles()
 
-        # pull out colorbar arguments
-        self.colorbar = kwargs.get('colorbar')
+    def _add_colorbar(self,
+                      mappable=None,
+                      cborientation="horizontal",
+                      cbshrink=0.75,
+                      cbpad=0.11,
+                      cbdrawedges=True,
+                      cbticks=None,
+                      cbticklabels=None,
+                      cbextend="neither",
+                      cbticklabelsize=None):
```

Comment:
`cbticklabelsize` and `cbticklabelsize` are not actually being used, should either be implemented or removed. Could be added to a todo as a placeholder in comments if you desire